Xcode analyzer issues reported

### Description
Running the Xcode analyzer in combination with integrating Bugsnag via SPM results in 3 Xcode analyzer issues.  Our particular CI pipelines runs Xcode analyzer as part of our build workflow so ideally there should be no issues reported.

### Issue

These are the analyzer issues being reported by Xcode:

1. /bugsnag-cocoa/Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSMachHeaders.m:211:1: warning: Potential leak of memory pointed to by 'newImage'

2. /bugsnag-cocoa/Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSMach.c:345:18: warning: Although the value stored to 'kr' is used in the enclosing expression, the value is never actually read from 'kr'
            if ((kr = thread_suspend(thread)) != KERN_SUCCESS) {
                 ^    ~~~~~~~~~~~~~~~~~~~~~~
3. /bugsnag-cocoa/Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSMach.c:384:18: warning: Although the value stored to 'kr' is used in the enclosing expression, the value is never actually read from 'kr'
            if ((kr = thread_resume(thread)) != KERN_SUCCESS) {

### Environment

Library versions:

* Bugsnag version 6.1.2 via Swift Package Manager

### Example code snippet

See above

--- a/Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSMach.c
+++ b/Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSMach.c
@@ -347,6 +347,8 @@ bool bsg_ksmachsuspendAllThreadsExcept(thread_t *exceptThreads,
                 // Don't treat this as a fatal error.
                 BSG_KSLOG_DEBUG("thread_suspend (%08x): %s", thread,
                                 mach_error_string(kr));
+                // Suppress dead store warning when log level > debug
+                (void)kr;
             }
         }
     }
@@ -386,6 +388,8 @@ bool bsg_ksmachresumeAllThreadsExcept(thread_t *exceptThreads,
                 // Don't treat this as a fatal error.
                 BSG_KSLOG_DEBUG("thread_resume (%08x): %s", thread,
                                  mach_error_string(kr));
+                // Suppress dead store warning when log level > debug
+                (void)kr;
             }
         }
     }

--- a/Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSMachHeaders.m
+++ b/Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSMachHeaders.m
@@ -192,7 +192,7 @@ bool bsg_mach_headers_populate_info(const struct mach_header *header, intptr_t s
 void bsg_mach_headers_add_image(const struct mach_header *header, intptr_t slide) {
     
     BSG_Mach_Header_Info *newImage = malloc(sizeof(BSG_Mach_Header_Info));
-    if (newImage != NULL && errno != ENOMEM) {
+    if (newImage != NULL) {
         if (bsg_mach_headers_populate_info(header, slide, newImage)) {
             
             bsg_mach_headers_cache_lock();

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,13 @@
 Changelog
 =========
 
+## TBD
+
+### Bug fixes
+
+* Quiet some Analyzer false positives
+  [#789](https://github.com/bugsnag/bugsnag-cocoa/pull/789)
+
 ## 6.1.3 (2020-08-17)
 
 ### Bug fixes

